File: app/src/main/java/com/hjc/base/ui/senior/activity/ThemeActivity.java
package com.hjc.base.ui.senior.activity;

import android.content.res.Configuration;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatDelegate;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.hjc.base.R;
import com.hjc.base.constant.RoutePath;
import com.hjc.base.databinding.ActivityThemeBinding;
import com.hjc.base.utils.helper.RouteManager;
import com.hjc.baselib.activity.BaseMvmActivity;
import com.hjc.baselib.utils.helper.ActivityHelper;
import com.hjc.baselib.viewmodel.CommonViewModel;

@Route(path = RoutePath.URL_THEME)
public class ThemeActivity extends BaseMvmActivity<ActivityThemeBinding, CommonViewModel> {

    @Override
    protected int getLayoutId() {
        return R.layout.activity_theme;
    }

    @Override
    protected CommonViewModel getViewModel() {
        return null;
    }

    @Override
    protected void initData(@Nullable Bundle savedInstanceState) {
    }

    @Override
    protected void addListeners() {
        mBindingView.setOnClickListener(this);

        mBindingView.titleBar.setOnViewLeftClickListener(view -> finish());
    }

    @Override
    protected void onSingleClick(View v) {
        if (v.getId() == R.id.btn_switch) {
            int mode = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;

            if (mode == Configuration.UI_MODE_NIGHT_YES) {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
            } else if (mode == Configuration.UI_MODE_NIGHT_NO) {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
            }

            ActivityHelper.finishAllActivities();
            RouteManager.jumpWithTransition(RoutePath.URL_SPLASH, R.anim.fade_in, R.anim.fade_out, this);
            finish();
        }
    }
}

File: module-senior/src/main/java/com/hjc/module_senior/ui/theme/ThemeActivity.kt
package com.hjc.module_senior.ui.theme

import android.content.res.Configuration
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatDelegate
import com.alibaba.android.arouter.facade.annotation.Route
import com.gyf.immersionbar.ImmersionBar
import com.hjc.library_base.activity.BaseActivity
import com.hjc.library_base.utils.ActivityHelper
import com.hjc.library_base.viewmodel.CommonViewModel
import com.hjc.library_common.router.RouteManager
import com.hjc.library_common.router.RoutePath
import com.hjc.library_widget.bar.OnViewLeftClickListener
import com.hjc.module_senior.R
import com.hjc.module_senior.databinding.SeniorActivityThemeBinding

@Route(path = RoutePath.Senior.URL_THEME)
class ThemeActivity : BaseActivity<SeniorActivityThemeBinding, CommonViewModel>() {

    override fun getLayoutId(): Int {
        return R.layout.senior_activity_theme
    }

    override fun createViewModel(): CommonViewModel? {
        return null
    }

    override fun getImmersionBar(): ImmersionBar? {
        return ImmersionBar.with(this)
            .fitsSystemWindows(true)
            .statusBarColor(R.color.senior_color)
    }

    override fun initData(savedInstanceState: Bundle?) {

    }

    override fun addListeners() {
        mBindingView.onClickListener = this

        mBindingView.titleBar.setOnViewLeftClickListener(object : OnViewLeftClickListener {
            override fun leftClick(view: View?) {
                finish()
            }
        })
    }

    override fun onSingleClick(v: View?) {
        if (v?.id == R.id.btn_switch) {
            val mode: Int = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
            if (mode == Configuration.UI_MODE_NIGHT_YES) {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
            } else if (mode == Configuration.UI_MODE_NIGHT_NO) {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
            }
            ActivityHelper.finishAllActivities()
            RouteManager.jumpWithTransition(
                this,
                RoutePath.Main.URL_SPLASH,
                R.anim.base_fade_in,
                R.anim.base_fade_out
            )
            finish()
        }
    }
}